notebooks/Mitigating Disparities in Ranking from Binary Data.ipynb

```diff
@@ -0,0 +1,611 @@
+    "\n",
+    "quartiles_by_race = all_train_grouped[['lsat','ugpa']].quantile([.25, .50, .75]).rename(\n",
+    "    index={0.25: \"25%\", 0.5: \"50%\", 0.75: \"75%\"}, level=1).unstack()\n",
+    "\n",
+    "rates_by_race = all_train_grouped[['pass_bar']].mean().rename(\n",
+    "    columns={'pass_bar': 'pass_bar_rate'})\n",
+    "\n",
+    "summary_by_race = pd.concat([counts_by_race, quartiles_by_race, rates_by_race], axis=1)\n",
+    "display(summary_by_race)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The vast majority of the students in the study are white. There is a notable gap between white and black students in their incoming academic credentials: the 75th percentile of the LSAT scores of black students is lower than the 25th percentile of the LSAT scores among white students. There is a less severe, but still substantial gap in UGPA. The achievement gap is greatly diminished in terms of the bar passage rate (78% for black students and 97% for white students). The authors of the [LSAC study](https://eric.ed.gov/?id=ED469370) conclude that this justifies admission practices that look beyond LSAT and UGPA. Since we do not have access to such additional variables, in the remainder of this notebook we will seek to build a bar passage predictor from these two variables alone."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Unmitigated Predictor\n",
+    "\n",
+    "We first train a standard logistic regression predictor that does not seek to incorporate any notion of fairness."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.linear_model import LogisticRegression\n",
+    "\n",
+    "unmitigated_predictor = LogisticRegression(solver='liblinear', fit_intercept=True)\n",
+    "unmitigated_predictor.fit(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We view the probabilistic predictions produced by the logistic model as scores and evaluate their accuracy in separating positive examples from the negative ones (i.e., the students who pass the bar from those who do not). Because of a high imbalance between positive and negative examples (the overall pass rate for the bar exam is over 94%), our accuracy metric is the [area under the ROC curve](https://en.wikipedia.org/wiki/Receiver_operating_characteristic#Area_under_the_curve) (AUC). It is equal to the probability that a randomly chosen positive example is scored above a randomly chosen negative example. An AUC of 0.5 means that the evaluated scores are no better than a random coin flip, whereas AUC of 1.0 means that the evaluated scores are perfect. The AUC metric has two desirable properties: (1) it is not sensitive to the imbalance between positives and negatives, and (2) it is preserved by monotone transformations of the score.\n",
```

Comment:
Maybe I'm not up-to-date on the latest approach, but I thought AUC is what we're looking at, whereas the algorithm optimizes for a different metric. If that's the case we should make that very clear or people are going to be very confused about what the algorithm does vs. what we're looking at.

Reply:
I think this is the root of my confusions over this notebook. We are assessing models based on a metric they weren't asked to optimise. I understand that the highly imbalanced dataset is going to give a lot of trouble, but it's not surprising that the AUC scores are bad when the estimators weren't trying to optimise for it.

Reply:
I've added some clarification. i'm going to merge and we can revise in future once we include "balanced error" as the objective in mitigation techniques (or if we run into any confusion from users).